models/utilities/default_model_mod.f90

```diff
@@ -283,6 +302,157 @@ subroutine pert_model_copies(state_ens_handle, ens_size, pert_amp, interf_provid
 
 end subroutine pert_model_copies
 
+!--------------------------------------------------------------------
+
+!> Parses the character table that was read in from
+!> model_nml:model_state_variables and returns a
+!> state_var_type state_vars with nvars ; netcdf variable names ;
+!> qtys (kinds) ; clamp values ; updates
+!> that there are valid entries for the dart_kind.
+!
+!> Verifies that the namelist entry was filled in correctly, and checks
+!> that there are valid entries for the dart_kind.
+
+function parse_variables_clamp(vars_table) result(state_vars)
+
+character(len=*), intent(in) :: vars_table(MAX_STATE_VARIABLE_FIELDS_CLAMP)
+type(state_var_type) :: state_vars
+
+character(len=NF90_MAX_NAME) :: netcdf_var_name, dart_qty_str, update
+character(len=256) :: string1, string2
+integer :: i, ivar
+
+! Loop through the variables array to get the actual count of the number of variables
+do ivar = 1, MAX_STATE_VARIABLES
+   ! If the first element in the row is an empty string, the loop has exceeded the extent of the variables
+   if (vars_table(5*ivar-4) == '') then
+      state_vars%nvars = ivar-1
+      exit
+   endif
+enddo
```

Comment:
if  state_var%nvars == MAX_STATE_VARIABLES 

then MAX_STATE_VARIABLES might not be big enough.

Reply:
@mjs2369 do you think an error E_ERR message here?
That way if people have >= 100 state variables this is caught here and they can increase MAX_STATE_VARIABLES as needed?

Reply:
@hkershaw-brown yes definitely, good catch

Reply:
sounds good. I'll put the error message in and we can do the release tomorrow!

Reply:
this actually leaves state_vars%nvars as unset

Reply:
so you don't get to this point if you have > 100 entries in the namelist. You get an invalid namelist entry:

message:  INVALID NAMELIST ENTRY:                       'TEMP_CUR33 ', 'QTY_POTENTIAL_TEMPERATURE', 'NO_COPY_BACK', in namelist model_nml